game-core/src/test/java/games/strategy/engine/message/RemoteActionCodeTest.java

```diff
@@ -0,0 +1,49 @@
+package games.strategy.engine.message;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.lang.reflect.Method;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.aggregator.AggregateWith;
+import org.junit.jupiter.params.aggregator.ArgumentsAccessor;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregationException;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregator;
+import org.junit.jupiter.params.provider.CsvFileSource;
+
+public class RemoteActionCodeTest {
+
+  @ParameterizedTest
+  @CsvFileSource(resources = "/required-op-codes.txt")
+  void verifyCorrectOpCode(
+      final int opCode, @AggregateWith(MethodAggregator.class) final Method method) {
+    final var remoteActionCode = method.getAnnotation(RemoteActionCode.class);
+
+    assertThat("No annotation present for " + method, remoteActionCode, is(notNullValue()));
```

Comment:
Why is this expected to be true? Can that be clarified in the assertion comment somehow perhaps?
Perhaps we can rephrase to say something like `"Expected an annotation to present on method:  "  + method`, it's generally clearer to explain what and why you were expected rather than to rephrase the error condition. EG: `result was not 4` vs `Expected 2+2 to be 4`

Reply:
Well `Method#getAnnotation(Class)` returns null if no annotation of the specified type is available on the method.
The tests set obviously only contains the methods that are being used over the network, so they are required to have this annotation.

Perhaps it would help if we said `Method %s not annotated with @RemoteActionCode`?

Reply:
> The tests set obviously only contains the methods that are being used over the network, so they are required to have this annotation.

I think that might depend on the maintainers context, revisiting this at first, that may not obvious until you spend some time investigating that. Telling a maintainer directly that to begin with saves time, makes code more scalable as this could be one of a dozen tests or tasks that a person has to go through.

The updated message example is an improvement, I like it :+1: